cmd/notation/cert/cleanupTest.go

```diff
@@ -0,0 +1,97 @@
+package cert
+
+import (
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/notaryproject/notation-go/config"
+	"github.com/notaryproject/notation-go/dir"
+	"github.com/notaryproject/notation/cmd/notation/internal/truststore"
+	"github.com/notaryproject/notation/internal/osutil"
+	"github.com/spf13/cobra"
+)
+
+type certCleanupTestOpts struct {
+	keyName string
+}
+
+func certCleanupTestCommand(opts *certCleanupTestOpts) *cobra.Command {
+	if opts == nil {
+		opts = &certCleanupTestOpts{}
+	}
+	command := &cobra.Command{
+		Use:   "cleanup-test <common_name>",
+		Short: "[Experimental] Clean up test key and its corresponding self-signed certificate created by the generate-test command.",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("missing certificate common_name")
+			}
+			opts.keyName = args[0]
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return cleanupTestCert(opts)
+		},
+	}
+	return command
+}
+
+func cleanupTestCert(opts *certCleanupTestOpts) error {
+	keyName := opts.keyName
+	var finalError []error
+	err := config.LoadExecSaveSigningKeys(func(keys *config.SigningKeys) error {
+		keySuite, err := keys.Get(keyName)
+		if err != nil {
+			return err
+		}
+		if keySuite.ExternalKey != nil {
+			return errors.New("cleanup-test can only apply to non-external keys created by the generate-test command")
+		}
+		if keySuite.X509KeyPair == nil {
+			return errors.New("cleanup-test requires key pair files created by the generate-test command")
+		}
+		// delete the key file from keyPath
+		keyPath := keySuite.X509KeyPair.KeyPath
+		err = osutil.DeleteFile(keyPath)
+		if err != nil {
+			finalError = append(finalError, fmt.Errorf("cannot delete the key file from %q during cleanup-test, %v", keyPath, err))
```

Comment:
Why not print the error directly?

Reply:
We may don't need to print an error if the file doesn't exist.

Reply:
This PR and its related issue are pending discussions. Will address comments once everyone agrees on the design.

Reply:
As we discussed offline, there are, in total, 4 deletions/removals in cleanup-test. We want all deletions/removals to be run even some of them may fail during this process. Eventually, we will print out all the succeeded operations followed by all the failed cases. If we return the error directly, then the user may need to call `notation cert cleanup-test` multiple times to remove all the related files which is not very user friendly.